src/WaipuData.cpp

```diff
@@ -201,6 +201,8 @@ void WaipuData::LoginThread()
       }
 
       continue;
+    } else if (m_login_status == WAIPU_LOGIN_STATUS::INVALID_CREDENTIALS) {
+       kodi::QueueNotification(QUEUE_ERROR, "", kodi::addon::GetLocalizedString(30030));
```

Comment:
Can you confirm it is okay to call this function from the login thread?

Reply:
I am not sure. Have you runtime tested? Did it work?

Reply:
On x64 desktop it works. But I don't have other devices to test on.

Reply:
Okay, I also have only x64 linux. But the changed code does not look platform specific. Also, the CI build for different OS. Should be fine. I runtime tested and your code works an improves the user experience. Thanks! I will release soon